src/parser.js

```diff
@@ -255,7 +255,16 @@ const grammar = {
             ['. SYMBOL ( arguments )', code`method.$2(/*@2*/current, $4)`],
             ['.. SYMBOL', code`fn.recursive(/*@2*/current, "$2")`],
             ['..( block )', code`fn.recursive(current, current => { $2 })`],
-            ['.[ block ]', code`fn.filter(current, current => { $2 })`]
+            ['.[ block ]', code`fn.filter(current, current => { $2 })`],
+            ['[ e : e ]', code`fn.slice(current, $2, $4, 1)`],
+            ['[ e : e : e ]', code`fn.slice(current, $2, $4, $6)`],
+            ['[ : e ]', code`fn.slice(current, 0, $3, 1)`],
+            ['[ : e : e ]', code`fn.slice(current, 0, $3, $5)`],
+            ['[ e : ]', code`fn.slice(current, $2, current.length, 1)`],
```

Comment:
My suggestion is to pass `undefined` instead of `current.length` and convert `undefined` to `current.length` inside `slice()` when it's appropriate.